src/plugins/mer/merplugin.cpp

```diff
@@ -168,6 +174,57 @@ bool MerPlugin::initialize(const QStringList &arguments, QString *errorString)
             MerQmlLiveBenchManager::startBench);
     menu->addAction(startQmlLiveBenchCommand);
 
+    Utils::MacroExpander *expander = Utils::globalMacroExpander();
+    expander->registerVariable("mer:merssh", tr("Location of merssh"),
+                               []()
+    {
+        return QString(Core::ICore::libexecPath() + QLatin1String("/merssh") + QStringLiteral(QTC_HOST_EXE_SUFFIX));
+    });
+    expander->registerVariable("mer:sshport", tr("SSH port"),
+                               []()
+    {
+        Kit* kit = SessionManager::startupProject()->activeTarget()->kit();
+        if (MerSdkManager::isMerKit(kit))
+        {
+            MerSdk* sdk = MerSdkKitInformation::sdk(kit);
+            return QString::number(sdk->sshPort());
+        }
+        return QString();
+    });
+    expander->registerVariable("mer:privatekeyfile", tr("Private key file"),
+                               []()
+    {
+        Kit* kit = SessionManager::startupProject()->activeTarget()->kit();
+        if (MerSdkManager::isMerKit(kit))
+        {
+            MerSdk* sdk = MerSdkKitInformation::sdk(kit);
+            return sdk->privateKeyFile();
+        }
+        return QString();
+    });
+    expander->registerVariable("mer:sharedhome", tr("Shared home path"),
+                               []()
+    {
+        Kit* kit = SessionManager::startupProject()->activeTarget()->kit();
+        if (MerSdkManager::isMerKit(kit))
+        {
+            MerSdk* sdk = MerSdkKitInformation::sdk(kit);
+            return sdk->sharedHomePath();
+        }
+        return QString();
+    });
+    expander->registerVariable("mer:sshuser", tr("SSH user name"),
+                               []()
+    {
+        Kit* kit = SessionManager::startupProject()->activeTarget()->kit();
+        if (MerSdkManager::isMerKit(kit))
+        {
+            MerSdk* sdk = MerSdkKitInformation::sdk(kit);
+            return sdk->userName();
+        }
+        return QString();
+    });
+    
```

Comment:
These are already available via "%{CurrentBuild:Env:NAME}", so in the external tool definition one can use e.g. MER_SSH_PORT="%{CurrentBuild:Env:MER_SSH_PORT}" to add the environment variable.